### ISSUE TITLE
How to show multiple data to respective multiselect(dropdown)?

### ISSUE THREAD

**Issue by maryheng** [monterail/vue-multiselect#459]
Hi all,

So the expected behaviour, when the page first loads, i want to get data from the server to display in respective dropdowns. However, i'm unable to display in the respective Multiselect, dropdown, because i'm using v-for to generate the rows of data.

I have this rows of items details that is created using v-for.

And this is the data I want to put into each dropdown.

```
"displayStaff": [
    {
      "staffId": "e35a49d8-2011-4644-abad-ef316fcfe3c2",
      "name": "Katy Perry"
    },
    {
      "staffId": "e6fee6f0-2d2c-4a3d-8f93-c1e03d08a9e7",
      "name": "Mary Heng"
    },
    {
      "staffId": "e6fee6f0-2d2c-4a3d-8f93-c1e03d08a9e7",
      "name": "Mary Heng"
    },
    {
      "staffId": "e35a49d8-2011-4644-abad-ef316fcfe3c2",
      "name": "Katy Perry"
    }
  ]
```

However, I can't seem to get each json object into the respective dropdown when the page first uploads.

This is my code for Multiselect, dropdown:
```
      <multiselect
        :id="item"
        :value="displayStaff"
        :options="allStaffs"
        :searchable="false"
        :allow-empty="false"
        :show-labels="false"
        label="name"
        track-by="name"
        @input="updateStaffData"
        @open="openStaffOptions"
        >
      </multiselect>  
```

Here is the reproduced jsfiddle:
https://jsfiddle.net/maryheng/34geecr0/1/

**Comment by Zip753** [comment]
Hi @maryheng,

Can you please elaborate a little bit more on what is supposed to happen?

1. In your fiddle you have three identical `v-for` blocks and only one item in each block, is it supposed to be so? Or maybe what you really want is one `v-for` with three items?
2. Also, do you want to have same options in all dropdowns? If yes, you could just update `options` when you receive data from the server and just pass the same options for all multiselects.
3. You're using `:key="item"` in your `v-for` block, which is probably not what you want. Instead I recommend that you assign an ID to each `item` (in other words: add an ID property to each `item` object or use existing property as an ID). That way you can safely add `value` field to each `item` and use it with multiselect, like this:
    ```html
    <div v-for="item in itemArray" :key="item.id">
      ...
      <multiselect :value="item.value" ...></multiselect>
    </div>
    ```
    and `itemArray` will be of this shape:
    ```javascript
    itemArray: [{ id: 1, value: null, ... }, { id: 2, value: null, ... }, ...]
    ```
4. You're passing `displayStaff` array as `value` to multiselects, but you don't have multiple choice enabled. So either you want to enable it or to pass a single object as a `value`.

**Comment by maryheng** [comment]
Hi @Zip753 , sorry for the confusion. Maybe I will explain the background to be more clear. 

Basically, I have an update page which looks something like this:
<img width="1366" alt="screen shot 2017-08-07 at 9 43 19 pm" src="https://user-images.githubusercontent.com/13627091/29029209-7d21aafc-7bb9-11e7-9c50-224a66ae580d.png">


So first to generate those item details, I used a` v-for` to generate rows of input fields and the dropdown from multiselect. I'm able to get data from the array and bind it to the respective input fields. However, I do not know how to bind respective data to the respective multiselect dropdowns.

Basically, the data I get back from the server dyanmically is `itemArray`, and i want to use the staffId in the `itemArray` to cross-reference to the `displayStaff` to get the name and then display the name respectively to the respective dropdowns. 

Hence, the expected behaviour would show, **1st dropdown would display = Katy, 2nd dropdown would display = Selena, 3rd dropdown would display = Susan.** 

Updated fiddle: https://jsfiddle.net/maryheng/34geecr0/5/

Is there a way I can do this? Thank you for your time!

**Comment by Zip753** [comment]
Thanks for the description, it's seems to be much more clear now, though I still don't see how `options` and `displayStaff` are different.

As far as I understood, for each item you want to associate the item's multiselect with the value from `displayStaff` that has the matching ID. Since multiselect's value is one of the options, you'll have to find it yourself, the simplest way that I can think of (not sure if the best actually) is to define a method that would return a value from `displayStaff` that matches the item's `staffId`, for example like this:
```javascript
methods: {
  findValueForItem (item) {
    return this.displayStaff.find(i => i.staffId === item.staffId)
  }
}
```
and then bind `findValueForItem(item)` to multiselect's `value`. Now you also have to update `staffId` whenever you change the value, this can be achieved by using the custom handler, like this:
```javascript
methods: {
  updateItemWithValue (item, selectedOption) {
    item.staffId = selectedOption.staffId
  }
}
```
and then using it as an event handler via `@input="updateItemWithValue(item, $event)"` (I wish there was a cleaner way to do it; `$event` corresponds to the selected option in this case).

Here's the updated fiddle: https://jsfiddle.net/zip753/34geecr0/6/

P.S. I really recommend that you change your `key` in `v-for`, it's potentially dangerous and Vue will scream in the console until you change it.

**Comment by maryheng** [comment]
Hi @Zip753 , sorry for the delayed reply. Thank you for your advice and explanation! I'm able to understand and port it over to my project and it works too! Also, took your advice in changing the `key` in `v-for`. Thank you for your help 👍
